terraform/charm/main.tf

```diff
@@ -0,0 +1,87 @@
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = ">= 0.16.0"
+    }
+  }
+}
+
+resource "juju_application" "haproxy" {
+  name  = var.haproxy_application_name
+  model = var.juju_model_name
+  units = var.haproxy_units
+
+  charm {
+    name     = "haproxy"
+    revision = var.haproxy_charm_revision
+    channel  = var.haproxy_charm_channel
+    base     = var.haproxy_charm_base
+  }
+
+  config = var.haproxy_config
+
+  expose {}
+}
+
+resource "juju_application" "grafana-agent" {
```

Comment:
You should move this resource to the "product" definition

Reply:
thanks, moved and updated the resource name